### PR TITLE
Adding support for Cloudflare Token API.

### LINKED ISSUE #3080: [RFE]: Use cloudflare API token instead of global key
#### Feature request type
I'd like cloudflare.conf to use a specific API key that only has access to the firewall instead of giving the global API key.

#### Description
There are two ways to access CloudFlare's APIs : The first is the "quick and dirty" way, with a global API key that has access to everything in your account. It is essentially equivalent to your password.
The second way is by using API Tokens, that are made specifically for one application and only has the required permissions. As example, the token for fail2ban would only have the permission to read and write firewall rules.

I have no idea if it is actually feasible to do in fail2ban as you probably need to make multiple requests

### PR DESCRIPTION
**Pull request by Logic-32**
Closes #3080

Copied mostly from cloudflare.conf and adapted for zone-based Token permissions.  I've been running this for awhile now on my server.  It works but, I'm the only one with access to my server so outside of the manual testing I did I can't say how bulletproof it is.

Before submitting your PR, please review the following checklist:

- [x] **CHOOSE CORRECT BRANCH**: if filing a bugfix/enhancement
      against certain release version, choose `0.9`, `0.10` or `0.11` branch,
      for dev-edition use `master` branch
- [x] **CONSIDER adding a unit test** if your PR resolves an issue
- [x] **LIST ISSUES** this PR resolves
- [x] **MAKE SURE** this PR doesn't break existing tests
- [x] **KEEP PR small** so it could be easily reviewed.
- [x] **AVOID** making unnecessary stylistic changes in unrelated code
- [x] **ACCOMPANY** each new `failregex` for filter `X` with sample log lines
      within `fail2ban/tests/files/logs/X` file

### CODE REVIEW

File: config/action.d/cloudflare-token.conf [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+cftarget = ip6
+
+# The firewall mode Cloudflare should use.  Default is "block" (deny access).
+# Consider also "js_challenge" or other "allowed_modes" if you want.
+#
+cfmode = block
+
+# The message to include in the firewall IP banning rule.
+#
+notes = Fail2Ban
```
Comment (sebres) [inline config/action.d/cloudflare-token.conf:92]: Wouldn't it be better to use jail name in notes? Similar to: https://github.com/fail2ban/fail2ban/blob/f81f85569d88da74d76f286b431dc54d0ae00748/config/action.d/cloudflare.conf#L47
Just for the case the same IP gets banned in two jails at the same time, so the ticket is grouped by `notes` (would be used as an additional identifier during unban).

Comment (Logic-32) [reply config/action.d/cloudflare-token.conf]: Done.

Comment (sebres) [reply config/action.d/cloudflare-token.conf]: Why did you delete the section `[Init?family=inet6]`? Doesn't cloudflare expect target `ip6` anymore for IPv6?

Comment (sebres) [reply config/action.d/cloudflare-token.conf]: Oh right, sorry for noise.

**Comment by Logic-32** [comment]
@sebres, is there anything else you need from me to get this merged?

**Pull request merged**